extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts

```diff
@@ -76,21 +78,25 @@ export class EncryptedMsgMailFormatter extends BaseMailFormatter {
       const mimeEncodedPlainMessage = await Mime.encode(msgBody, { Subject: newMsg.subject }, atts);
       const encryptedMessage = await SmimeKey.encryptMessage({ pubkeys: x509certs, data: Buf.fromUtfStr(mimeEncodedPlainMessage) });
       const body = { "encrypted/buf": Buf.fromUint8(encryptedMessage.data) };
-      return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body, type: 'smimeEncrypted', atts: [], isDraft: this.isDraft });
+      //return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body, type: 'smimeEncrypted', atts: [], isDraft: this.isDraft });   
+      return await SendableMsg.createSMime(this.acctEmail, this.headers(newMsg), body, { isDraft: this.isDraft });
```

Comment:
This can be simplified further - the createSMime method specifically can accept `encryptedMessage.data` directly, and convert that into `{ "encrypted/buf": Buf.fromUint8(encryptedMessage.data) }` internally since that is the only way to create an smime message.

Reply:
Ok